### PR TITLE
Add EASM .NET data plane SDK

### PR DESCRIPTION
**Pull request by thang-bit**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by pallavit** [comment]
Given this is a new SDK - could you please share who has been helping you with the onboarding? Also could you add the correct owners to the https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS file.

/cc: @jsquire

**Comment by thang-bit** [comment]
@pallavit I have been working on this .NET sdk by myself

**Comment by thang-bit** [comment]
Hi @KrzysztofCwalina can you please review this PR?

File: sdk/easm/Azure.Analytics.Defender.Easm/perf/Azure.Analytics.Defender.Easm.Perf.csproj
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/perf/Azure.Analytics.Defender.Easm.Perf.csproj:1]: If we're not going to create performance tests, let's remove this folder, as it serves no purpose.

File: sdk/easm/Azure.Analytics.Defender.Easm/stress/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/stress/README.md:1]: If you're not creating stress tests, let's remove this folder as it serves no purpose.

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:17]: We should tweak this to avoid mentioning SDK versions that are end-of-life.   At this point, only `net6.0` and higher are supported, so just mentioning that we need the ".NET SDK" would be fine.

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:40]: This needs content

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:138]: These sections need content

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/README.md:12]: This page needs content

Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/README.md:12]:
```suggestion
# Azure.Analytics.Defender.Easm samples
```

The Microsoft style guide requires sentence-style casing for headings

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample3_DiscoveryTemplates.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample3_DiscoveryTemplates.md:4]:
```suggestion
```
nit: Extra blank

File: sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj:10]: These suppressions will need to be removed unless the archboard grants an exception for the latter two.

File: sdk/easm/Azure.Analytics.Defender.Easm/tsp-location.yaml
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/tsp-location.yaml:3]: This must point to the main repository.

**Review by jsquire — CHANGES_REQUESTED** [review]

File: sdk/easm/Azure.Analytics.Defender.Easm/perf/Azure.Analytics.Defender.Easm.Perf.csproj
Comment (thang-bit) [reply sdk/easm/Azure.Analytics.Defender.Easm/perf/Azure.Analytics.Defender.Easm.Perf.csproj:1]: I tried to remove the folder, but it threw exception Azure.Analytics.Defender.Easm.Perf.csproj when I tried to generate the sdk again. So, I'm leaving it here for now

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:17]: We should remove the phrase ".NET Core", as the branding is just ".NET" now.   I'd also remove the last line, as there are no supported SDKs where the C# 8 syntax is in preview.

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:43]:
```suggestion

Defender EASM includes the discovery of the following kinds of assets:
```

Adding one comment here, but all header sections should be followed by a blank line, according to the Microsoft style guide.

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:153]:
```suggestion
## Troubleshooting

```

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:158]: Add content or remove the section.

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md:1]: The Microsoft style guide requires that headers use sentence-casing rather than the title-casing used here.  Please adjust all headers in markdown files.

File: sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj:3]: This is the description that will show up in the NuGet overview.   We should consider updating this, as the current wording doesn't tell someone unfamiliar with the library what it should be used for or what it is.

File: sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj:10]: The comment does not map to this warning.   Why are we suppressing a warning about a project file unable to be loaded?  That seems wrong.

File: sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj:4]: We should fix this.  That's not the title we want on NuGet.

File: sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/src/Azure.Analytics.Defender.Easm.csproj:6]: This should be a set of keyword tags for NuGet to aid with search discoverability.

File: sdk/easm/Azure.Analytics.Defender.Easm/src/properties/AssemblyInfo.cs
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/src/properties/AssemblyInfo.cs:6]: If we're not using this, let's remove it.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS]: There's a bunch of things that need to be fixed for this:

- It is in the wrong place; please add this above in the SDK section, in alphabetical order.

- The form of this is incorrect.   There should be a seprarate PR and ServiceLabel section.  Please follow the existing entries as examples.

- Your path is too general.  You should have the specific library that you're responsible for in the path segment.  There will likely be other libraries under `/sdk/easm/` owned by different teams.

- You cannot just add a service label; these are centrally managed.  Please send email to "SDKGitHubNinjas" and request a new label.  Once approved, it will be added to the central list and pushed to all repositories.

Comment (thang-bit) [reply .github/CODEOWNERS]: Do I use PRLabel instead of ServiceLabel here?

Comment (thang-bit) [reply .github/CODEOWNERS]: This is the list of our approved names: https://github.com/Azure/azure-sdk/issues/5062. Maybe it can help us name the label?

**Comment by thang-bit** [comment]
@jsquire if we don't get the name for PR label by tomorrow, can we merge this PR without the code owners for now? I can have a follow up PR to add the code owner. I plan to release the SDK beta version on next week 03/05. 
I checked the release check list, the code owner isn't the list: https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/15/Doc.aspx?sourcedoc=%7Bad5cabd9-978a-4ae5-9a84-f4ab0ebdd76e%7D&action=edit&wd=target(Guidelines.one%7C2d826c9f-fb1d-4676-ada3-636c9afe09ef%2FRelease%20checklist%20for%20.NET%7C383e2aa7-e765-46c7-a735-99d4628465c6%2F)&wdorigin=NavigationUrl&share=IgHZq1ytipflSpqE9KsOvdduATYSaLtsZtX5JTiWByYsw1M

I also updated the CHANGELOG.md

**Comment by thang-bit** [comment]
@jsquire, I have updated the PRLabel and ServiceLabel to "Defender EASM", do you know why the test is still failing saying that there's no "Defender EASM" in this repository?

**Comment by jsquire** [comment]
> @jsquire, I have updated the PRLabel and ServiceLabel to "Defender EASM", do you know why the test is still failing saying that there's no "Defender EASM" in this repository?

That will continue to happen until tomorrow when the linter's sync job processes changes from each repository.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:482]:
```suggestion
/sdk/easm/Azure.Analytics.Defender.Easm/                           @thang-bit
```

The path must end with a slash.

File: sdk/easm/Azure.Analytics.Defender.Easm/CHANGELOG.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/CHANGELOG.md:4]:
```suggestion

- Defender Azure EASM (External Attack Surface Management) client library for Java. This package contains Microsoft Azure EASM client library.
```

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:55]:
```suggestion
### [Discovery][discovery_documentation]

```

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:58]:
```suggestion
#### [Discovery Groups][discovery_groups_documentation]

```

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:66]:
```suggestion

<!-- CLIENT COMMON BAR -->
```

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:155]:
```suggestion
### Enabling logging

```

The Microsoft style guide requires sentence-style casing.

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/README.md:7]: seems like there should be another of these for a more specific match.   Is `defender` a thing in MS Docs?

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/README.md:80]: I had been ignoring "Assets" and "Discovery Group" text as proper names, but the samples seem to lower case these, so we should follow sentence-casing here as well, per the Microsoft style guide.

Leaving one comment, but all the headers in this area should be adjusted.

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md:50]:
```suggestion
```
nit : extra space

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md:59]:
```suggestion
```

nit: extra space

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample3_DiscoveryTemplates.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample3_DiscoveryTemplates.md:29]:
```suggestion
```

nit: Extra space

File: sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample1_DiscoveryGroups.md
Comment (jsquire) [inline sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample1_DiscoveryGroups.md:18]:
```suggestion
```

nit: extra space

**Review by jsquire — COMMENTED** [review]
There's a few minor formatting nits left, but otherwise, we're good once the linter job runs tomorrow morning.

**Review by jsquire — COMMENTED** [review]
Let's please fix the title casing in the main README to conform to the [style guide](https://learn.microsoft.com/style-guide/capitalization#sentence-style-capitalization-in-titles-and-headings).  Since these assets are published to MS Learn, they need to conform to those standards.

**Comment by jsquire** [comment]
> @jsquire I think you might need refresh browser/ clear caches. git pull request failed sync some of pushed commits

Yup.  That looks right now.

**Review by jsquire — APPROVED** [review]

File: sdk/easm/Azure.Analytics.Defender.Easm/README.md
Comment (thang-bit) [reply sdk/easm/Azure.Analytics.Defender.Easm/README.md]: I already pushed this change earlier, it should show up in a bit

**Comment by jsquire** [comment]
@thang-bit :  Looks like pushes with changes have slowed.  If you're ready to merge, please let me know and I'll hit the button.

**Comment by thang-bit** [comment]
@jsquire The PR is ready to merge

**Pull request merged**
